### DISCUSSION TITLE
wiki/vcal/acgbday/issue/

### DISCUSSION THREAD

**Discussion by giscus** [Vanadiry/blog-comments#19] [blog-comments]
# wiki/vcal/acgbday/issue/

宇宙虽有其起源，却没有终结；星球亦有起源，却因其自身之力走向毁灭。这里由Vanadiry用希望和魔法创建，在孤独且混沌的轮回中寻找梦想和奇迹。

https://magic.vanadiry.com/wiki/vcal/acgbday/issue/

**Comment by M1zuon** [comment]
关于自动化爬取的可行性 
萌娘百科有按生日分类的页面，或许可以做到人物姓名自动爬取？[分类:按生日分类](https://mzh.moegirl.org.cn/Category:%E6%8C%89%E7%94%9F%E6%97%A5%E5%88%86%E7%B1%BB)

**Reply by Vanadiry** [reply]
感谢建议！

现在在测试自动调用番组计划API来获取角色信息的程序，还不太完善，半自动化，且仍需手动校对。
获取角色生日方面，这个API会更方便些，可以实现提供一个动画的id，获取这部动画中出现的角色及其大部分信息。
由于番组计划上角色描述大多是官方给到的，且部分无翻，所以萌娘百科主要用来搜集角色描述这类信息。为了尽量给到详细的描述并且避免剧透，这部分工作目前只能人工(｡ ́︿ ̀｡)

比如[鹿目圆](https://bgm.tv/character/10439)角色页，可以获取到像下面这样的内容：
```json
{
    "birth_mon": 10,
    "gender": "female",
    "birth_day": 3,
    "name": "鹿目まどか",
    …
}
```
